test/generators/UserAnswersGenerator.scala

```diff
@@ -36,7 +36,7 @@ trait UserAnswersGenerator extends TryValues {
 
     Arbitrary {
       for {
-        id      <- nonEmptyString
+        id      <- nonEmptyString.map(MovementReferenceNumber(_))
```

Comment:
We can use the `Arbitrary[MovementReferenceNumber]`
```suggestion
        id      <- arbitrary[MovementReferenceNumber]
```

Reply:
Good spot, thanks :+1: